stdlib/Std/src/FFI/C/Value.luna

```diff
@@ -0,0 +1,445 @@
+class CString:
```

Comment:
As we're going to be using the `foreign` namespace for things that aren't FFI as well, can we change this module to be `Std.Foreign.C.Value`?

Reply:
In general, I don't like the naming here, so this is not something to change for now, but a note to our future selves to make these into plain `String`, `UInt`, etc, once we can `import Std.Foreign.C.Value as C` to disambiguate.

Reply:
Additionally, I have some concerns that we're using `Int`, `UInt`, `Long`, etc, as these types aren't rigorously defined by the C standard and depend on ABI. In general I would much rather see specific numeric types as follows:

- `(U)Int8`
- `(U)Int16`
- `(U)Int32`
- `(U)Int64`

By making our types ABI dependent rather than specific sizes we run the risk of colliding with bad assumptions about types on different platforms.

Reply:
1. Sure, we can put this file under `Foreign`
2. Yes, after we can disambiguate by named imports, let's change it to just `String` etc. For now we need to stick to `CInt` etc. as otherwise it's not usable due to conflicts with Base.
3. I feel like we need both and planned to add `(U)IntN` types later on. But I feel that we have to use things like `Int` precisely for their being ABI dependent. Since the same C code using `int`s can be compiled for both 32 and 64 bit platform, this is the only way to ensure some kind of portability.

Reply:
This is a good point, too, but I think they are potentially more important than being added 'later on'. How much work would it be to add them for the first release of the FFI?

Reply:
We'll add the (U)IntXX types before merging